### ISSUE TITLE
How should I specify the downloadUrl?

### ISSUE THREAD

**Issue by qq253498229** [nwutils/nw-builder#808]
### Issue Type

- [ ] Bug Report
- [x] Feature Request
- [ ] Other

### Current/Missing Behaviour

I'm a Chinese developer. I can't download from https://dl.nwjs.io because of the network environment. 
It has been stuck at 8MB and does not support breakpoint reconnect.

Fortunately, Alibaba has a mirror site in China: `https://npmmirror.com/mirrors/nwjs/` and `https://npm.taobao.org/mirrors/nwjs/` .

Now I don't know how to configure this site address.

I tried to specify using the `downloadUrl` parameter, but it didn't work:

```
nwbuild({
  srcDir: "./temp/*",
  outDir: outDir,
  cacheDir: "./cache", // "./cache" | string
  mode: "build", // "run" | "build"
  version: version, // "latest" | "stable" | string
  flavor: "normal", // "normal" | "sdk"
  platform: platform, // "linux" | "osx" | "win"
  arch: arch, // "ia32" | "x64" | "arm64"
  downloadUrl: 'https://npmmirror.com/mirrors/nwjs',
})
```

And I also tried to modify the source code in `nw-builder/src/get/download.js` , but it still doesn't work:

28th line
```
  let url;
  let out;
  return new Promise((res, rej) => {
    if (downloadUrl === "https://dl.nwjs.io" || downloadUrl === "https://npmmirror.com/mirrors/nwjs") {
      url = `${downloadUrl}/v${version}/nwjs${
        flavor === "sdk" ? "-sdk" : ""
      }-v${version}-${platform}-${architecture}.${
        platform === "linux" ? "tar.gz" : "zip"
      }`;
      out = resolve(cacheDir, `nw.${platform === "linux" ? "tar.gz" : "zip"}`);
    } 
```

### Additional Info

- Package version:
- Operating System:
- Node version:
- NW.js version:
- Repro link:
- ...

**Comment by tharatau** [comment]
The npmmirror link errors out:
```xml
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<RequestId>63FF7C45B9DB8531312AFCA7</RequestId>
<HostId>npmmirror.oss-cn-shanghai.aliyuncs.com</HostId>
<Key>binaries/nwjs</Key>
<EC>0026-00000001</EC>
</Error>
```

Looks like the mirror download link format is slightly different and is not supported. This gives me a 404 error:
`https://registry.npmmirror.com/mirrors/nwjs/v0.73.0/nwjs-sdk-v0.73.0-linux-x64.tar.gz`

**Comment by qq253498229** [comment]
the link is :
https://npmmirror.com/mirrors/nwjs/v0.73.0/nwjs-sdk-v0.73.0-linux-x64.tar.gz

**Comment by tharatau** [comment]
Ahh my bad, typing from my phone. I'm currently afk. What you tried out should work. What's the response/error you get?

**Comment by tharatau** [comment]
I was able to reproduce the issue even after allow listing the mirrors.

**Comment by tharatau** [comment]
This has been fixed in v4.1.0-beta.3
